src/enc.rs

```diff
@@ -77,6 +91,35 @@ impl MulAssign for Enc {
     }
 }
 
+impl Sub for Enc {
+    type Output = Self;
+
+    #[inline]
+    fn sub(self, rhs: Self) -> Self::Output {
+        Self {
+            inner: self.inner - rhs.inner,
+        }
+    }
+}
+
+impl SubAssign for Enc {
+    #[inline]
+    fn sub_assign(&mut self, rhs: Self) {
+        *self = *self - rhs
```

Comment:
Does this also work if we're encrypting/decrypting `u32`? Or is there some trick to it?